Forta-Bot-Challenge-4/COMP03/src/agent.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+} from "forta-agent";
+import { ethers } from "forta-agent";
+import { SUPPLY_RATE_ABI, SUPPLY_RATE_CONTRACT_ADDRESS } from "./constants";
+
+import { getSupplyRate, checkSupplyRate } from "./utils";
+
+export function provideHandleTransaction(
+  provider: ethers.providers.Provider,
+  supplyRateABI: string,
+  blockNumber: number
```

Comment:
The block number is not updated inside the function, so it will always be 0.

Also, please note that most of @RCantu92's comments on COMP01 apply here as well.